openedx_tagging/core/tagging/models/base.py

```diff
@@ -344,11 +345,11 @@ def _check_taxonomy(
         Subclasses can override this method to perform their own taxonomy validation checks.
         """
         # Must be linked to this taxonomy
-        return object_tag.taxonomy_id and object_tag.taxonomy_id == self.id
+        return bool(object_tag.taxonomy_id) and object_tag.taxonomy_id == self.id
```

Comment:
I'm unsure if it applies here, but in the edgy case where `taxonomy_id=0`, `bool(object_tag.taxonomy_id)` will mistakenly return `False`.
```suggestion
        return object_tag.taxonomy is not None and object_tag.taxonomy_id == self.id
```

Reply:
I don't think we really need to worry about id being zero, but it's better to be safe than sorry, so I've made that change :)